presto-hive/src/main/java/com/facebook/presto/hive/WrapperJobConf.java

```diff
@@ -0,0 +1,294 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapred.JobConf;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+
+import static java.util.Objects.requireNonNull;
+
+public class WrapperJobConf
+        extends JobConf
+{
+    protected final Configuration config;
+
+    public WrapperJobConf(Configuration config)
+    {
+        super();
+        this.config = requireNonNull(config, "config is null");
+    }
+
+    @Override
+    public Iterator<Map.Entry<String, String>> iterator()
+    {
+        if (config == null) {
```

Comment:
I don't see a no-arg constructor for this class.
If config is passed in as null, then `requireNonNull` would catch it.
Wondering why we need a config null check everywhere.?

Also can you help me understand why config should be a protected var instead of private.
You can add a getter in this class for config and then remove the direct reference to config from `CachingJobConf`
for a similar example see how `DeleteNode` constructor takes in a `sourceLocation` and `id` , calls `super` and in the `replaceChildren()` method uses the getters `getSourceLocation()` , `getId()` .